src/compiler/checker.ts

```diff
@@ -138,6 +138,7 @@ namespace ts {
                 node = getParseTreeNode(node, isExportSpecifier);
                 return node ? getExportSpecifierLocalTargetSymbol(node) : undefined;
             },
+            getExportSymbolOfSymbol: getExportSymbolOfValueSymbolIfExported,

```

Comment:
My apologies. `getExportSymbolOfValueSymbolIfExported` will only get the export symbol if the symbol is a Value symbol, not a Type symbol.

Perhaps this might be more reliable:

```ts
function getExportSymbolOfSymbol(symbol: Symbol) {
  return getMergedSymbol(symbol.exportSymbol || symbol);
}
```

Otherwise, the example in your comment in _types.ts_ isn't strictly valid.